src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -1,73 +1,114 @@
 import os
 import time
-import numpy as np
 import mgrs
 import logging
 import mimetypes
-from osgeo import gdal, osr
-from pyproj import CRS
 import h5py
 import glob
-import mgrs
+import ast
+import rasterio
+import numpy as np
+import geopandas as gpd
 import datetime
+
+from osgeo import gdal, osr
+from pyproj import CRS
+from rasterio.warp import transform_bounds
+from shapely.geometry import Polygon
+
 from dswx_sar.version import VERSION as software_version
 from dswx_sar import dswx_sar_util
 from dswx_sar.dswx_runconfig import _get_parser, RunConfig
 
 
 logger = logging.getLogger('dswx_s1')
 
-def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name, verbose=False):
-    """Get bounding box cooridnate from MGRS tile
+def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name,
+                                             mgrs_db_path=None):
+    """Get geographic extents from given MGRS tile
     Parameters
     ----------
     mgrs_tile_name: str
-        MGRS tile grid ID (e.g. 10SEG)
+        Name of MGRS tile (ex. 18LWQ)
+    mgrs_db_path : str
+        path for MGRS database
+
     Returns
     -------
-    x_value_min: float
+    minx: float
         minimum x cooridate (UTM) for given MGRS tile
-    x_value_max: float
+    maxx: float
         maximum x cooridate (UTM) for given MGRS tile
-    y_value_min: float
+    miny: float
         minimum y cooridate (UTM) for given MGRS tile
-    y_value_max: float
+    maxy: float
         maximum y cooridate (UTM) for given MGRS tile
+    epsg: int
+        epsg code
     """
-    mgrs_obj = mgrs.MGRS()
-    lower_left_utm_coordinate = mgrs_obj.MGRSToUTM(mgrs_tile_name)
-    utm_zone = lower_left_utm_coordinate[0]
-    is_northern = lower_left_utm_coordinate[1] == 'N'
-    x_min = lower_left_utm_coordinate[2]
-    y_min = lower_left_utm_coordinate[3]
-
-    # compute boundaries
-    x_list = []
-    y_list = []
-    for offset_x_multiplier in range(2):
-        for offset_y_multiplier in range(2):
-            x = x_min + offset_x_multiplier * 109.8 * 1000
-            y = y_min + offset_y_multiplier * 109.8 * 1000
-            x_list.append(x)
-            y_list.append(y)
-
-    x_value_min = np.min(x_list)
-    x_value_max = np.max(x_list)
-    y_value_min = np.min(y_list)
-    y_value_max = np.max(y_list)
-
-    return x_value_min, x_value_max, y_value_min, y_value_max
-
-def save_mgrs_tile(source_tif_path,
-                   output_dir_path,
-                   output_tif_name,
-                   output_mgrs_id,
-                   method='nearest',
-                   no_data_value=255):
+    # if mgrs tile database is not given,
+    # return the coordinates based on python mgrs
+    if mgrs_db_path is None:
```

Comment:
Please, consider separating this function into two. One that uses the `mgrs` library and another that uses the MGRS DB.